usrsctplib/user_environment.c

```diff
@@ -82,6 +86,34 @@ read_random(void *buf, size_t size)
 	return;
 }
 
+void
+finish_random(void)
+{
+	return;
+}
+#elif defined(SCTP_USE_OPENSSL_RAND)
+#include <stdlib.h>
+#include <openssl/rand.h>
+
+void
+init_random(void)
+{
+	return;
+}
+
+void
+read_random(void *buf, size_t size)
+{
+	if (!RAND_bytes((uint8_t *)buf, size)) {
+#ifdef INVARIANTS
+		panic("RAND_bytes failed.");
+#else
+		abort();
```

Comment:
I don't see abort called outside of an INVARIANTS block anywhere else, but failing to retrieve a random number seems severe enough a reason to abort.

This shouldn't normally happen, as openssl seeds the random number generator automatically by default, but it seemed right to remain cautious.

Reply:
Hmm. I asked initially a couple of consumers of the lib and the result was that the lib should not terminate the program (or write anything to stdout/stderr). The other implementations of `read_random()` retry until they are successful. Can't you do the same here?

Reply:
All right, got rid of the `abort`.